flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericCatalogTable.java

```diff
@@ -33,30 +35,53 @@
 public class GenericCatalogTable implements CatalogTable {
 	// Schema of the table (column names and types)
 	private final TableSchema tableSchema;
-	// Properties of the table
-	private final Map<String, String> properties;
 	// Statistics of the table
 	private final TableStats tableStats;
+	// Partition keys if this is a partitioned table. It's an empty set if the table is not partitioned
+	private final List<String> partitionKeys;
+	// Properties of the table
+	private final Map<String, String> properties;
 	// Comment of the table
 	private String comment = "This is a generic catalog table.";
 
-	public GenericCatalogTable(TableSchema tableSchema, TableStats tableStats, Map<String, String> properties) {
-		this.tableSchema = tableSchema;
-		this.tableStats = tableStats;
+	public GenericCatalogTable(
+		TableSchema tableSchema,
```

Comment:
We need to indent 2 tabs here, so you don't need to leave a blank line in the beginning of this method.

Reply:
ok